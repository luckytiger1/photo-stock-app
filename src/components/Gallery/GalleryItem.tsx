import React from 'react';
import { withRouter } from 'react-router-dom';
import downloadIcon from './assets/download.svg';
import heartIcon from './assets/heart.svg';
import resizeIcon from './assets/resize.svg';
import UserInfo from '../UserInfo/UserInfo';
import DownloadBtn from '../DownloadBtn/DownloadBtn';
import LikeBtn from '../LikeBtn/LikeBtn';

const GalleryItem = ({ image, updateFavorites, history, match }: any) => {
  return (
    <div className="photo-item" key={image.id}>
      <img
        src={image.urls.small}
        alt={image.alt_description && 'stock-img'}
        className="photo-item__img"
      />
      {match.path === '/:photoId' ? null : (
        <>
          <UserInfo image={image} />
          <div className="photo-item__info">
            <LikeBtn
              itemClass="info__btn"
              icon={heartIcon}
              addToFavorites={() => updateFavorites(image)}
            />

            <button
              className="info__btn"
              type="button"
              onClick={() => history.push(image.id)}
            >
              <img src={resizeIcon} alt="resize" />
            </button>
            <DownloadBtn
              itemClass="info__btn"
              link={image.links.download}
              icon={downloadIcon}
            />
          </div>
        </>
      )}
    </div>
  );
};

export default withRouter(GalleryItem);
