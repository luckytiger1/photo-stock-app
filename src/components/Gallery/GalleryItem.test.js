import React from 'react';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import { GalleryItem } from './GalleryItem';

let mockImage;

beforeEach(() => {
  expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
  mockImage = {
    id: 'Shj3j4',
    urls: {
      small: 'image.png',
    },
    links: {
      download: 'downloadImg.com',
    },
  };
});

describe('GalleryItem', () => {
  it('<GalleryItem /> renders all info if match path is not photoId', () => {
    const mockMatch = {
      path: '/favorites',
    };
    expect(
      shallow(<GalleryItem match={mockMatch} image={mockImage} />),
    ).toMatchSnapshot();
  });
  it("<GalleryItem /> doesn't render all info if match path is photoId", () => {
    const mockMatch = {
      path: '/:photoId',
    };
    expect(
      shallow(<GalleryItem match={mockMatch} image={mockImage} />),
    ).toMatchSnapshot();
  });
  it('should render self and subcomponents', () => {
    const mockMatch = {
      path: '/favorites',
    };
    const mockAddToFavorites = {
      updateFavorites: jest.fn(),
    };
    const mockHistory = ['a', 'b'];
    const enzymeWrapper = shallow(
      <GalleryItem
        match={mockMatch}
        image={mockImage}
        updateFavorites={mockAddToFavorites.updateFavorites}
        history={mockHistory}
      />,
    );
    expect(enzymeWrapper.find('LikeBtn')).toHaveLength(1);
    enzymeWrapper
      .find('LikeBtn')
      .props()
      .addToFavorites({ id: '1', url: 'image1.png' });
    expect(mockAddToFavorites.updateFavorites.mock.calls.length).toBe(1);
    enzymeWrapper.find('button').simulate('click');
    expect(mockHistory.length).toBe(3);
  });
});
