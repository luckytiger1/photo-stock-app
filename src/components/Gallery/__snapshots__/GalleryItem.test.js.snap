// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem <GalleryItem /> doesn't render all info if match path is photoId 1`] = `
<div
  className="photo-item"
  key="Shj3j4"
>
  <img
    className="photo-item__img"
    src="image.png"
  />
</div>
`;

exports[`GalleryItem <GalleryItem /> renders all info if match path is not photoId 1`] = `
<div
  className="photo-item"
  key="Shj3j4"
>
  <img
    className="photo-item__img"
    src="image.png"
  />
  <UserInfo
    image={
      Object {
        "id": "Shj3j4",
        "links": Object {
          "download": "downloadImg.com",
        },
        "urls": Object {
          "small": "image.png",
        },
      }
    }
  />
  <div
    className="photo-item__info"
  >
    <LikeBtn
      addToFavorites={[Function]}
      icon="heart.svg"
      itemClass="info__btn"
    />
    <button
      className="info__btn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="resize"
        src="resize.svg"
      />
    </button>
    <DownloadBtn
      icon="download.svg"
      itemClass="info__btn"
      link="downloadImg.com"
    />
  </div>
</div>
`;
